packages/contracts/contracts/optimistic-ethereum/OVM/bridge/messaging/OVM_L1CrossDomainMessenger.sol

```diff
@@ -76,9 +87,50 @@ contract OVM_L1CrossDomainMessenger is iOVM_L1CrossDomainMessenger, Abs_BaseCros
      * Public Functions *
      ********************/
 
+    /**
+     * Pause relaying.
+     */
+    function pause()
+        external
+        onlyOwner
+    {
+        _pause();
+    }
+
+    /**
+     * Block a message.
+     * @param _xDomainCalldataHash Hash of the message to block.
+     */
+    function blockMessage(
```

Comment:
These should emit an event so that the state of `blockedMessages` can be determined from the set of emitted events

Reply:
These good?

Reply:
Looks good to me! I see some tradeoffs between having 2 events vs 1 event with 2 args, the second arg being an enum that determines if it was blocked or allowed. I don't have really strong opinions but am wondering if you have put any thought into the tradeoffs

Reply:
> I don't have really strong opinions but am wondering if you have put any thought into the tradeoffs

I considered one event with another argument. I went this way because I tend to prefer a function/event with a name that makes its purpose obvious. The alternative would be something like the following which I find to be less explicit. 

```
event MessageStatus(
        bytes32 indexed _xDomainCalldataHash,
        enum isAllowed
 );
 ```
 
Your question also prompted me to look for some code that monitors for events. 
In `watcher.ts`. I think this approach is more consistent with the code watching for [`SentMessages(bytes)`](https://github.com/ethereum-optimism/optimism/blob/946c7a281a3216487bc71875b3491b9bab1e385c/packages/core-utils/src/watcher.ts#L59), and [``RelayedMessage(bytes32)`](https://github.com/ethereum-optimism/optimism/blob/946c7a281a3216487bc71875b3491b9bab1e385c/packages/core-utils/src/watcher.ts#L80), and seems like it makes the typescript more readable as well.